### DISCUSSION TITLE
Implement a post processor that takes more than one variables

### DISCUSSION THREAD

**Discussion by pengfejeac** [idaholab/moose#25347] [Q&A Modules: General] [answered]
Hi there,

I've searched around but could't find anything related. I'm trying to do some post processing that involves more than one variables. So I start implementing my own post processor based on ElementIntegralArrayVariablePostprocessor and hope to add parameters so that I can input more variables. Some suggestions I found claims that "_sys.getFieldVariable" is a good idea, but I can't get it working in my own post processors, it works well in customized kernels.

I'm still new to Moose, any help will be much appreciated!
Peter

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can add as many variable parameters as you need by adding them as coupled variables:
In validParams:
```
params.addCoupledVar("var_name1", default_var1_name, "first coupled variable (description)");
params.addCoupledVar("var_name2", default_var2_name, "second coupled variable (description)");
```

in the constructor:
```
_var1(coupledValue("var_name1")),
_var2(coupledValue("var_name2")),
```

in the header:
```
const VariableValue & _var1;
const VariableValue & _var2;
```

there's numerous examples if you search for any of these routines in the code base
Guillaume
